### PR TITLE
Add functional tests for Send to Device form on Android page

### PR DESCRIPTION
**Pull request by alexgibson**
Note: this is currently blocked from merging due to the form failing on dev. I've filed a bug which can hopefully be resolved: https://bugzilla.mozilla.org/show_bug.cgi?id=1235534

### CODE REVIEW

**Comment by alexgibson** [comment]
@davehunt r?

File: tests/functional/firefox/test_android.py [hunk 1]
```diff
@@ -15,6 +16,29 @@ def test_family_navigation(base_url, selenium):
...
+    page = AndroidPage(base_url, selenium).open()
+    with pytest.raises(TimeoutException):
+        page.send_to_device.click_send()
+
+
+@pytest.mark.smoke
+@pytest.mark.nondestructive
+def test_send_to_device_not_supported_locale(base_url, selenium):
+    page = AndroidPage(base_url, selenium, 'it').open()
+    assert not page.send_to_device.is_displayed
```
Comment (alexgibson) [inline tests/functional/firefox/test_android.py]: Note to self: I should also assert that the Play Store badge is visible here.

File: tests/functional/firefox/test_android.py [hunk 2]
```diff
@@ -15,6 +16,30 @@ def test_family_navigation(base_url, selenium):
     assert page.family_navigation.is_menu_displayed
 
 
+@pytest.mark.flaky(reruns=1)
```
Comment (davehunt) [inline tests/functional/firefox/test_android.py]: Do we have evidence that this is flaky or is it just using the same feature as the tests already marked as flaky?

Comment (alexgibson) [inline tests/functional/firefox/test_android.py]: As it submits to basket I'm making the assumption we might get the same issue as we see on other newsletters.

File: tests/functional/firefox/test_android.py [hunk 3]
```diff
@@ -15,6 +16,30 @@ def test_family_navigation(base_url, selenium):
...
+def test_send_to_device_fails_when_missing_required_fields(base_url, selenium):
+    page = AndroidPage(base_url, selenium).open()
+    with pytest.raises(TimeoutException):
+        page.send_to_device.click_send()
+
+
+@pytest.mark.smoke
+@pytest.mark.nondestructive
+def test_send_to_device_not_supported_locale(base_url, selenium):
+    page = AndroidPage(base_url, selenium, 'it').open()
```
Comment (davehunt) [inline tests/functional/firefox/test_android.py]: Nit: If you include the keyword for the argument it makes it more readable `locale='it'`

**Comment by davehunt** [comment]
r+

**Comment by alexgibson** [comment]
Removing blocker as the form is now fixed on dev

**Pull request merged**